src/app/teamSheets/page.tsx

```diff
@@ -0,0 +1,55 @@
+import { redirect } from 'next/navigation'
+
+import { faUsersRectangle } from '@fortawesome/free-solid-svg-icons'
+
+import { getKindeServerSession } from '@kinde-oss/kinde-auth-nextjs/server'
+
+import prisma from '@db/prismaSingleton'
+
+import { findOrCreateUser } from '@functions/user'
+
+import PageHeader from '@components/PageHeader'
+
+import type { TeamSheetWithTeam } from '@types'
+
+import TeamSheetListItem from './_TeamSheetListItem'
+
+const Page = async () => {
+  const { getUser } = getKindeServerSession()
+  const kindeCurrentUser = await getUser()
+
+  const currentUser = await findOrCreateUser(kindeCurrentUser)
+  const { id: currentUserId } = currentUser
```

Comment:
getting a ts warning here because `currentUser` can be null, easiest fix is to move the redirect up

```
const currentUser = await findOrCreateUser(kindeCurrentUser)
if (!currentUser) redirect('/#')
```